unstructured/partition/pdf.py

```diff
@@ -464,7 +464,52 @@ def convert_pdf_to_images(
             yield image
 
 
-def add_pytesseract_bbox_to_elements(elements, bboxes, width, height):
+def _get_element_box(
+    element: Text,
+    boxes: List[str],
+    box_idx: int,
+    char_count: int,
+):
+    """Helper function to get the bounding box of an element.
+
+    Args:
+        element
+        boxes (List[str], optional): Defaults to [].
+        box_idx (int, optional): Defaults to 0.
+        char_count (int, optional): Defaults to 0.
+    """
+    min_x = float("inf")
+    min_y = float("inf")
+    max_x = 0
+    max_y = 0
+    box_text = ""
+    for box in boxes[box_idx : box_idx + char_count]:  # noqa
+        char, _x1, _y1, _x2, _y2, _ = box.split()
+
+        # pytesseract cleans some characters from text that still appear in the bounding boxes,
+        # so those bounding boxes should be skipped
+        if char not in element.text:
+            char_count += 1
+            continue
```

Comment:
so `char` here might still be not in the `element`. e.g., if the actual text should have been "apple vs. oranges" and `element.text` is `aple vs. oranges`, it is missing the second `p`. Then this logic will still pass because there is still a `p` in the text. More realistically think about a long page it likely has all the most common characters so even if a whole word gets cleaned out the individual characters can still be in `element.txt`. And actually even whole word may still fall into this kind of trap.

Reply:
Yah, I realize it's not the best check. It was from when I was initially trying to figure out what to do with the random '~' characters that were showing up. I think this might actually be handled by the `while` loop I later added to compare the first char of element.text and the bounding box char.

Reply:
so maybe we can remove this logic then?